DNN Platform/Library/Data/DataProvider.cs

```diff
@@ -3317,14 +3317,14 @@ public virtual IDataReader GetFolderMappingSettings(int folderMappingID)
 
         #region Password History
 
-        public virtual IDataReader GetPasswordHistory(int userId)
+        public virtual IDataReader GetPasswordHistory(int userId, int passwordsRetained, int daysRetained)
```

Comment:
This creates compatibility issues. This must be deprecated and a new method created.

Reply:
In the past I had been told that `DataProvider` was not intended to be backwards compatible, but I'm happy to keep compatibility, if that's desired.  However, I won't be able to turn any attention to this until at least next week.

Reply:
That's right and the ideal case, but in practice, some third party modules might have used these. We need to try keeping backward compatibility as much as possible.